tests/core/filtering/test_contract_on_event_filtering.py

```diff
@@ -1,9 +1,26 @@
 import pytest
 
+from eth_utils import (
+    is_address,
+)
+
 # Ignore warning in pyethereum 1.6 - will go away with the upgrade
 pytestmark = pytest.mark.filterwarnings("ignore:implicit cast from 'char *'")
 
 
+@pytest.mark.parametrize('call_as_instance', (True, False))
+def test_create_filter_address_parameter(web3, emitter, Emitter, call_as_instance):
+    if call_as_instance:
+        event_filter = emitter.events.LogNoArguments.createFilter(fromBlock="latest")
+    else:
+        event_filter = Emitter.events.LogNoArguments.createFilter(fromBlock="latest")
+
+    if call_as_instance:
+        # Assert this is a single string value, and not a list of addresses
+        assert isinstance(event_filter.filter_params['address'], str)
+        assert is_address(event_filter.filter_params['address'])
```

Comment:
Do these asserts need to be in the `if` check? Right now it's not asserting anything if `call_as_instance` is `False`.

Reply:
If call_as_instance is False, I think the contract is not deployed, so there should be no address field in the parameters.  I suppose I should assert that as well.

Reply:
...but wait.  If the class contract is an undeployed contract with no address, how are transactions sent.  :confused:

Reply:
because you cant.

```
(Pdb) Emitter.address
(Pdb) emitter.address
'0xF2E246BB76DF876Cef8b38ae84130F4F55De395b'
(Pdb)
(Pdb) Emitter.functions.logString("hi").transact()
*** ValueError: Please ensure that this contract instance has an address.
(Pdb) 
```

worldview restored. :relieved: